src/plugins/score-plugin-fx/Fx/ValueFilter.hpp

```diff
@@ -0,0 +1,340 @@
+#pragma once
+#include <Engine/Node/SimpleApi.hpp>
+#include <libossia/3rdparty/dno/DeNoiser.hpp>
+#include <ossia/network/value/format_value.hpp>
+
+namespace Nodes::ValueFilter
+{
+using namespace dno;
+
+struct Node
+{    
+  struct NoiseFilter
+  {
+    NoiseFilter()
+      : dno_i{DeNoiser<int>{}},
+        dno_v{DeNoiser<double>{},
+              DeNoiser<double>{},
+              DeNoiser<double>{},
+              DeNoiser<double>{}}
+    {}
+
+    DeNoiser<int> dno_i;
+    std::vector<DeNoiser<double>> dno_v;
+
+    ossia::value filter(const ossia::value& val)
+    {
+      struct vis
+      {
+        ossia::value operator()() const { return {}; }
+        ossia::value operator()(const ossia::impulse&) const { return {}; }
+        ossia::value operator()(int i) const
+        {
+          return nFilt->dno_i(i);
+        }
+        ossia::value operator()(float f) const
+        {
+          return nFilt->dno_v[0](f);
+        }
+        ossia::value operator()(bool b) const
+        {
+          return b;
+        }
+        ossia::value operator()(const std::string& s) const
+        {
+          return s;
+        }
+        ossia::value operator()(const ossia::vec2f& t) const
+        {
+          return ossia::make_vec(nFilt->dno_v[0](t[0]),
+              nFilt->dno_v[1](t[1]));
+        }
+        ossia::value operator()(const ossia::vec3f& t) const
+        {
+          return ossia::make_vec(nFilt->dno_v[0](t[0]),
+              nFilt->dno_v[1](t[1]),
+              nFilt->dno_v[2](t[2]));
+        }
+        ossia::value operator()(const ossia::vec4f& t) const
+        {
+          return ossia::make_vec(nFilt->dno_v[0](t[0]),
+              nFilt->dno_v[1](t[1]),
+              nFilt->dno_v[2](t[2]),
+              nFilt->dno_v[3](t[3]));
+        }
+        ossia::value operator()(const std::vector<ossia::value>& t) const
+        {
+          return t;
+        }
+
+        NoiseFilter* nFilt;
+
+        vis(NoiseFilter* parent)
+          : nFilt{parent}
+        {}
+      };
+
+      try {
+        return ossia::apply(vis{this}, val.v);
+      } catch(std::exception& e) {
+        std::cerr << e.what() << std::endl;
```

Comment:
if you can, use `ossia::logger().error("{}", e.what());` instead so that it shows in the message log